tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
+    this.promise = new Promise((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+  }
+  addEvent(eventData?: any) {
+    this.eventData = [
+      ...this.eventData,
+      eventData
+    ];
+    if (this.eventData.length >= this.expectedEvents) {
+      this.resolve(this.eventData);
+    }
+  }
+  reset(expectedEvents?: number) {
+    this.expectedEvents = expectedEvents || this.expectedEvents;
```

Comment:
FWIW, it might be cleaner to have a .waitFor(events: number): Promise method instead of this (and drop the number from the constructor).

And you could potentially have a .waitForCount(count: number) and .waitForEventMatching(condition: (event: T => boolean)) versions depending on whether you want to wait for a specific number, or wait until a condition is true, etc.